### PR TITLE
Remove the parts which ByteBuf's 'hasArray()' method is used

### PR DESCRIPTION
**Pull request by aldemirenes**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,6 +44,19 @@
...
+          <archive>
+            <manifest>
+              <mainClass>fully.qualified.MainClass</mainClass>
+            </manifest>
+          </archive>
+          <descriptorRefs>
+           <descriptorRef>jar-with-dependencies</descriptorRef>
+          </descriptorRefs>
+        </configuration>
+      </plugin>
```
Comment (BenFradet) [inline pom.xml]: You shouldn't have to need the assembly plugin?

Comment (aldemirenes) [reply pom.xml]: Since I put the nsq client to the es loader's repo for now, I used assembly plugin but I will remove it while I am adding CI/CD to project

File: project/BuildSettings.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  pomExtra := (
+    <developers>
+      <developer>
+        <name>Snowplow Analytics Ltd</name>
+        <email>support@snowplowanalytics.com</email>
+        <organization>Snowplow Analytics Ltd</organization>
+        <organizationUrl>http://snowplowanalytics.com</organizationUrl>
+      </developer>
+    </developers>)
+  )
```
Comment (BenFradet) [inline project/BuildSettings.scala]: indent

File: project/Dependencies.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+import sbt._
+
+object Dependencies {
+  val resolvers = Seq(
+    "Snowplow Analytics Maven releases repo" at "http://maven.snplow.com/releases/"
+  )
```
Comment (BenFradet) [inline project/Dependencies.scala]: why do we need it? I don't see any sp deps

File: src/main/java/com/snowplowanalytics/client/nsq/Connection.java [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-
-import com.github.brainlag.nsq.callbacks.NSQErrorCallback;
-import com.github.brainlag.nsq.exceptions.NSQException;
-import com.github.brainlag.nsq.exceptions.NoConnectionsException;
-import com.github.brainlag.nsq.frames.ErrorFrame;
-import com.github.brainlag.nsq.frames.MessageFrame;
-import com.github.brainlag.nsq.frames.NSQFrame;
-import com.github.brainlag.nsq.frames.ResponseFrame;
-import com.github.brainlag.nsq.netty.NSQClientInitializer;
+package com.snowplowanalytics.client.nsq;
```
Comment (BenFradet) [inline src/main/java/com/snowplowanalytics/client/nsq/Connection.java:1]: shouldn't we add an apache 2 headers to the source files? @alexanderdean

File: src/test/scala/com/snowplowanalytics/client/nsq/NSQConsumerTest.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
     //duration to wait before auto-requeing a message setting in nsqd, defined with -msg-timeout:
     //Set your timeout -msg-timeout="5s" command line when starting nsqd or changed this constant
     //to the default of 60000.
-    private static final long NSQ_MSG_TIMEOUT = 5000;
+    private static final long NSQ_MSG_TIMEOUT = 60000;
```
Comment (BenFradet) [inline src/test/scala/com/snowplowanalytics/client/nsq/NSQConsumerTest.java:19]: why was this changed?

File: src/test/scala/com/snowplowanalytics/client/nsq/NSQProducerTest.java [hunk 6]
```diff
@@ -150,33 +111,6 @@ public void testProduceOneMsgSsl() throws InterruptedException, NSQException, Ti
...
-            counter.incrementAndGet();
-            message.finished();
-        }, getSslAndSnappyConfig());
-        consumer.start();
-        while (counter.get() == 0) {
-            Thread.sleep(500);
-        }
-        assertEquals(1, counter.get());
-        consumer.shutdown();
-    }
```
Comment (BenFradet) [inline src/test/scala/com/snowplowanalytics/client/nsq/NSQProducerTest.java:177]: why were those tests removed?

File: src/main/java/com/snowplowanalytics/client/nsq/Connection.java (hunk 4 above)
Comment (alexanderdean) [reply src/main/java/com/snowplowanalytics/client/nsq/Connection.java:1]: No, it's MIT licensed (see https://github.com/brainlag/JavaNSQClient), I don't want to get into re-licensing the fork...

File: src/test/scala/com/snowplowanalytics/client/nsq/NSQConsumerTest.java (hunk 5 above)
Comment (aldemirenes) [reply src/test/scala/com/snowplowanalytics/client/nsq/NSQConsumerTest.java:19]: I wanted to change the value to default one for nsqd

File: src/test/scala/com/snowplowanalytics/client/nsq/NSQProducerTest.java (hunk 6 above)
Comment (aldemirenes) [reply src/test/scala/com/snowplowanalytics/client/nsq/NSQProducerTest.java:177]: There is a problem with Snappy option. I think it is related with change in the nsq but I could not find the source of the problem. Since we are not using snappy option for nsq in any project, I remove those tests temporarily. I forgot the open the issue this problem, will do it now. Is it okay if we make this release without snappy ?

Comment (BenFradet) [reply src/test/scala/com/snowplowanalytics/client/nsq/NSQProducerTest.java:177]: if it doesn't work let's remove support entirely and not just the tests.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+language: scala
+scala:
+- 2.11.5
```
Comment (BenFradet) [inline .travis.yml]: let's go with 2.11.11

File: project/BuildSettings.scala [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
```
Comment (BenFradet) [inline project/BuildSettings.scala]: @alexanderdean can we mix those apache 2 headers or should we remove them?

File: project/BuildSettings.scala [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  // Build settings for our app
+  lazy val buildSettings = Seq(
+    organization          :=  "com.snowplowanalytics"
+  )
+
+  lazy val publishSettings = bintraySettings ++ Seq(
+    publishMavenStyle := true,
+    publishArtifact := true,
+    publishArtifact in Test := false,
+    licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0.html")),
```
Comment (BenFradet) [inline project/BuildSettings.scala]: it's mit

**Review by BenFradet — COMMENTED** [review]
a couple of remarks regarding licensing

File: project/BuildSettings.scala (hunk 8 above)
Comment (alexanderdean) [reply project/BuildSettings.scala]: No, let's stick to MIT everywhere, good catch

**Review by BenFradet — APPROVED** [review]
LGTM :+1:
